ui-v3/src/services/accountBalanceService.ts

```diff
@@ -1,32 +1,211 @@
+      id: number,
+      asset: "ft" | "nft"
+   ) => {
+      let response;
+      try {
+         const { api } = this.getClientConfig(walletAddress);
+         let res;
+         if (asset === "ft") {
+            response = (
+               await axios.get(
+                  `${api}/metadata/v1/${asset}/${asset_identifiers}`
+               )
+            ).data;
+         } else {
+            response = (
+               await axios.get(
+                  `${api}/metadata/v1/${asset}/${asset_identifiers}/${id}`
+               )
+            ).data?.metadata;
+         }
+      } catch (error) {
+         console.log({ error });
+      }
+      return response;
+   };
+
+   private formatDecimals = (
+      value: number | string,
+      decimals: number,
+      isUmicro: boolean
+   ): string => {
+      if (isUmicro) {
+         return (Number(value) * 10 ** decimals).toFixed(0);
+      } else {
+         return (Number(value) / 10 ** decimals).toFixed(4);
       }
-    ];
-  }
+   };
+
+   async getSTXBalance(walletAddress: string, offset: number) {
+      const { api, chain } = this.getClientConfig(walletAddress);
+      const response = (
+         await axios.get(
+            `${api}/extended/v2/addresses/${walletAddress}/balances/stx?offset=${offset}`
+         )
+      )?.data;
+      const stxDecimal = 6;
+      const stxBalance = {
+         ...response,
+         image: "/stx-logo.svg",
+         decimals: stxDecimal,
+         actual_balance: this.formatDecimals(
+            response?.balance,
+            stxDecimal,
+            false
+         ),
+         name: "Stacks",
+         symbol: "STX",
+      };
+
+      return stxBalance;
+   }
+
+   async getNFTBalance(
+      walletAddress: string,
+      asset_identifiers: string,
+      offset: number
+   ): Promise<NFTBalanceResponse[]> {
+      try {
+         const { api } = this.getClientConfig(walletAddress);
+
+         const nftBalanceResponse = (
+            await axios.get(
+               `${api}/extended/v1/tokens/nft/holdings?principal=${walletAddress}&${
+                  asset_identifiers
+                     ? `asset_identifiers=${asset_identifiers}`
+                     : ""
+               }&offset=${offset}&tx_metadata=true`
+            )
+         )?.data?.results;
+
+         const nftBalance = await Promise.all(
+            nftBalanceResponse.map(async (res: nftResponseBalanceValues) => {
+               const id = res?.value?.repr?.replace("u", "");
+               const asset = res?.asset_identifier;
+               const tokenMeta = await this.handleGetMeta(
+                  walletAddress,
+                  asset?.split("::")[0],
+                  Number(id),
+                  "nft"
+               );
+               const tokenMetadata = await axios.get(tokenMeta?.token_uri);
+               return {
+                  asset_name: asset?.split("::")[1],
+                  asset_address: asset?.split("::")[0],
```

Comment:
That is asset_contract_id, isn't it? A fully qualified contract. Is that what we need? Do we need `contract_address` as well?